hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenMintHandler.java

```diff
@@ -173,11 +171,11 @@ private List<Long> mintNonFungible(
         validateFalse(metadata.isEmpty(), INVALID_TOKEN_MINT_METADATA);
 
         // validate token number from treasury relation
-        final var tokenId = asToken(treasuryRel.tokenNumber());
-        validateTrue(treasuryRel.tokenNumber() == token.tokenNumber(), FAIL_INVALID);
+        final var tokenId = treasuryRel.tokenId();
+        validateTrue(treasuryRel.tokenId().equals(tokenId), FAIL_INVALID);
```

Comment:
This is equivalent to a == a test.  I'm guessing it should have been an null check, or else the two lines are unnecessary.

Reply:
I'm guessing this should be `treasuryRel.tokenId().equals(token.treasuryAccountId())`